packages/dd-trace/src/appsec/iast/analyzers/intermendiate-cookies-analyzer.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+
+const Analyzer = require('./vulnerability-analyzer')
+const dc = require('../../../../../diagnostics_channel')
+const iastSetCookieChannel = dc.channel('datadog:iast:set-cookie')
+
+class IntermediateCookiesAnalyzer extends Analyzer {
```

Comment:
It could extend directly from `Plugin` since it's not using any method from `Analyzer` - just `addSub` which it's a proxy to the super class.

Reply:
From the foregoing, shouldn't this code live in `http` plugin?

Reply:
> It could extend directly from Plugin since it's not using any method from Analyzer - just addSub which it's a proxy to the parent class method.

Yes, changed

> From the foregoing, shouldn't this code live in http plugin?

I don't like the idea to add asm related code in the apm plugin. I want to prevent it if it is possible. And, the the plugins are configurable and they can be disabled.